File: src/main/java/ru/evotor/framework/core/action/event/cash_operations/CashOperationEvent.java
package ru.evotor.framework.core.action.event.cash_operations;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

import ru.evotor.IBundlable;

public abstract class CashOperationEvent implements IBundlable {
    private static final String KEY_DOCUMENT_UUID = "documentUuid";

    @NonNull
    private final String documentUuid;

    CashOperationEvent(@NonNull String documentUuid) {
        this.documentUuid = documentUuid;
    }

    @Nullable
    static String getDocumentUuid(@Nullable Bundle bundle) {
        return bundle == null ? null : bundle.getString(KEY_DOCUMENT_UUID);
    }

    @Override
    @NonNull
    public Bundle toBundle() {
        Bundle result = new Bundle();
        result.putString(KEY_DOCUMENT_UUID, documentUuid);
        return result;
    }

    @NonNull
    public String getDocumentUuid() {
        return documentUuid;
    }
}

File: src/main/java/ru/evotor/framework/kkt/event/CashOperationEvent.kt
package ru.evotor.framework.kkt.event

import android.os.Bundle

import ru.evotor.IBundlable
import ru.evotor.framework.getMoney
import java.math.BigDecimal

abstract class CashOperationEvent internal constructor(val documentUuid: String, val total: BigDecimal) : IBundlable {

    override fun toBundle(): Bundle {
        val result = Bundle()
        result.putString(KEY_DOCUMENT_UUID, documentUuid)
        result.putString(KEY_TOTAL, total.toPlainString())
        return result
    }

    companion object {

        private const val KEY_DOCUMENT_UUID = "documentUuid"

        private const val KEY_TOTAL = "total"

        internal fun getDocumentUuid(bundle: Bundle): String? = bundle.getString(KEY_DOCUMENT_UUID)

        internal fun getTotal(bundle: Bundle): BigDecimal? = bundle.getMoney(KEY_TOTAL)
    }
}
